sycl/source/detail/os_util.cpp

```diff
@@ -35,23 +40,23 @@ struct ModuleInfo {
   const char *Name;     // out
 };
 
-static int callback(struct dl_phdr_info *info, size_t size, void *data) {
-  unsigned char *Base = reinterpret_cast<unsigned char *>(info->dlpi_addr);
-  ModuleInfo *MI = (ModuleInfo *)data;
+static int callback(struct dl_phdr_info *Info, size_t Size, void *Data) {
+  unsigned char *Base = reinterpret_cast<unsigned char *>(Info->dlpi_addr);
+  ModuleInfo *MI = (ModuleInfo *)Data;
+  const unsigned char *TestAddr =
+      reinterpret_cast<const unsigned char *>(MI->VirtAddr);
 
-  for (int i = 0; i < info->dlpi_phnum; ++i) {
-    unsigned char *SegStart = Base + info->dlpi_phdr[i].p_vaddr;
-    unsigned char *SegEnd = SegStart + info->dlpi_phdr[i].p_memsz;
-    const unsigned char *TestAddr =
-        reinterpret_cast<const unsigned char *>(MI->VirtAddr);
+  for (int i = 0; i < Info->dlpi_phnum; ++i) {
+    unsigned char *SegStart = Base + Info->dlpi_phdr[i].p_vaddr;
+    unsigned char *SegEnd = SegStart + Info->dlpi_phdr[i].p_memsz;
 
     // check if the tested address is within current segment
     if (TestAddr >= SegStart && TestAddr < SegEnd) {
       // ... it is - belongs to the module then
       // dlpi_addr is zero for the executable, replace it
-      void *H = (void *)info->dlpi_addr;
+      void *H = (void *)Info->dlpi_addr;
```

Comment:
I think you can remove `(void *)`

Reply:
(void *) cannot be just removed here. Otherwise, the error is emitted due to conversion of 'long unsigned int' to 'void*'.
I replaced it with reinterpret_cast.

Reply:
Ah, right. I though `info->dlpi_addr` was an address already, so another kind of pointer.
Sorry about the noise...